org.eclipse.scout.rt.mom.api/src/main/java/org/eclipse/scout/rt/mom/api/AbstractMomTransport.java

```diff
@@ -34,7 +36,25 @@ public abstract class AbstractMomTransport implements IMomTransport {
 
   private static final Logger LOG = LoggerFactory.getLogger(AbstractMomTransport.class);
 
-  protected final FinalValue<IMomImplementor> m_delegate = new FinalValue<>();
+  protected final String m_momUid = UUID.randomUUID().toString();
+
+  private final Object m_delegateLock = new Object();
+  private volatile IMomImplementor m_delegate;
+
+  @Override
+  public String getId() {
+    return m_momUid;
+  }
+
+  @Override
+  public String getName() {
+    return getDelegate().getName();
+  }
+
+  @Override
+  public List<ISubscription> getSubscriptions() {
```

Comment:
A unit test getting (and checking) active subscriptions (+ reconnect) would be nice ;-)